### DISCUSSION TITLE
Accomodate 'external ducking' after wakeword

### DISCUSSION THREAD

**Discussion by SamJongenelen** [toverainc/willow#333] [Ideas]
background:
I have currently as a 'main driver' still a JBlink Google device. I use it for multiple things that are not yet part of either Willow nor HA's VoiceAssistant and it also has a nice audio as it is a complete (trojan) package. 
 I also have an M5 stack for Voice Assist ánd a S3 Box for Willow. Cool stuff to play with. 

My idea or question is related to 'ducking': when my household is using the Google device for music, the Willow device will still nicely react to the wakeword (depending on volume, of course) but, when it does, my audio on the Google home device will continue to play and mess with my voice interations. 

 So, as I am running HomeAssistant, is there any way I could react to a wakeword detection on a Willow device and 'duck' my audio device during this time?

**Comment by kovrom** [comment]
Had the same issue, so concocted a temp solution for myself, until willow folks will surely come up with something more efficient and interesting :)
Basically you can create an HA automation with Webhook trigger, where Webhook ID is your willow box hostname and then do If/Then with
```
IF: value template {{ trigger.data['key'] == 'wake' }}
Then: do whatever you want to do, lower the volume, pause, etc.
Else: Resume playback, increase the volume, etc
```
I do pause/resume, because it's much faster and willow is super fast, as you know :) , so volume change on HA part could be a bit slow to catch up. But pause/resume works right away. You can play around using my test fork of WAS for now: ghcr.io/kovrom/willow-application-server:duck
Make sure to set "Webhook URL" and enable "Send Webhook command on Wake" in WAS UI and restart WAS.
WAS sends  "wake" key on wake and "wake_end" on wake end.
